src/Standards/Generic/Tests/CodeAnalysis/UselessOverridingMethodUnitTest.1.inc

```diff
@@ -76,4 +87,56 @@ interface InterfaceFoo {
 
 trait TraitFoo {
     abstract public function sniffShouldBailEarly();
+
+    public function usefulMethodInTrait() {
+        parent::usefulMethodInTrait();
+
+        return 1;
+    }
+
+    public function uselessMethodInTrait() {
+        return parent::uselessMethodInTrait();
+    }
+}
+
+abstract class AbstractBar extends AbstractFoo {
+    public function uselessMethodInAbstractClass() {
+        parent::uselessMethodInAbstractClass();
+    }
+
+    public function usefulMethodInAbstractClass() {
+        $a = 1;
+        parent::usefulMethodInAbstractClass($a);
+    }
+}
```

Comment:
What is this testing ? (what isn't already tested)

Reply:
When adding the condition to bail early for functions that are not part of a class, anonymous class, or traits, I added this test to make sure that we had at least the basic tests covering abstract classes. It is true that abstract classes use the same token as regular classes, but I thought that it was still worth adding these tests just in case to avoid regressions in the future, even if, in this case, they are not very likely to happen.

Reply:
Abstract classes are treated no differently from normal classes by the sniff, abstract methods are, but this class doesn't even have an abstract method..., so it's not testing anything which isn't already being tested.

Reply:
Yes, I agree that it is not testing anything which isn't already being tested considering the current implementation of the sniff. I'm happy to remove those two tests if you prefer.

> but this class doesn't even have an abstract method

Only today, I realized that I could have added those two methods to the abstract class that already exists in the file. Does that sound better to you? If not, I can simply remove the tests.

Reply:
Feel free to move them into the existing abstract class. That would make more sense to me.

Reply:
Done